haskell-project.nix

```diff
@@ -66,6 +66,16 @@ in
             (lib.attrNames config.packages);
         withHoogle = true;
       };
+      modifyoverrides = s: name: value:
+        pkgs.haskell.lib.compose.overrideCabal
+          (drv: {
+            doCheck = ! value.disableTests;
+            enableLibraryProfiling = value.profiling;
+            enableExecutableProfiling = value.profiling;
```

Comment:
You might structure this in RFC42 style, so that knowledge about `overrideCabal` is transferable between it and haskell-flake.

Reply:
By 'RFC42' do you mean https://github.com/NixOS/rfcs/pull/42 ?